### DISCUSSION TITLE
whitelist a driver

### DISCUSSION THREAD

**Discussion by westyles** [HotCakeX/Harden-Windows-Security#189] [Q&A] [answered]
Hello.
Is there any way to whitelist a old driver that has been blocked WDAC by MS default?
The driver is now blocked because of a leaked certificate, and I don't want to remove all the blocking or testsignining ON mode.

**Comment by HotCakeX** [comment, marked as answer]
Hi,
The Microsoft recommended driver block rules are digitally signed by MSFT so we can't add supplemental policies to them since we don't have the private key of their certificate, so modifying it isn't an option. But we still have other options.

If you really want to run that vulnerable driver, first disable the block list in Microsoft defender -> Device Security

<img width="535" alt="image" src="https://github.com/HotCakeX/Harden-Windows-Security/assets/118815227/3d17d2e6-2979-406a-8bfb-7992ce2d89a2">

It's grayed out because of this

<img width="379" alt="image" src="https://github.com/HotCakeX/Harden-Windows-Security/assets/118815227/427834f6-edb4-427f-a220-15e6b927d06c">

and it will remain that way until you turn off Secure boot in your UEFI settings, then perform a restart, deactivate Memory Integrity, restart again, and then disable the vulnerable block list. The additional restarts are due to the UEFI lock mechanism.

Now you need to grab the block list [from here](https://learn.microsoft.com/en-us/windows/security/application-security/application-control/windows-defender-application-control/design/microsoft-recommended-driver-block-rules), save it as XML file, remove the rules responsible for blocking your driver, then convert it to .CIP file using [ConvertFrom-CIPolicy](https://learn.microsoft.com/en-us/powershell/module/configci/convertfrom-cipolicy) and deploy it. You can even generate a certificate using [Build-WDACCertificate](https://github.com/HotCakeX/Harden-Windows-Security/wiki/Build-WDACCertificate) and then deploy it using [Deploy-SignedWDACConfig](https://github.com/HotCakeX/Harden-Windows-Security/wiki/Deploy-SignedWDACConfig).

**Reply by westyles** [reply]
Thanks for the detailed answer!
It turns out that it can only be replaced with its own rules, first disabling security and those MS rules.
And after these changes, is it possible to re-enable secure boot or something else?

**Reply by HotCakeX** [reply]
Yes, you can enable secure boot again without turning on memory integrity. You can also set this policy to "Disabled" so it won't automatically be turned on later

<img width="1215" alt="image" src="https://github.com/HotCakeX/Harden-Windows-Security/assets/118815227/1a7082e4-9e43-420e-a4e5-5be5e93bf15b">
